### PR TITLE
Front-end for News and placeholder for Profile viewpager

### PR DESCRIPTION
**Pull request by rickwolox**
### Summary
Front end for News with loading  icon when refreshing, and at first instance no content

### Trello Card
https://trello.com/c/hpVnFC6i/35-u4-home-page-frontend-news

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/home/HomePageFragment.kt [hunk 1]
```diff
@@ -59,5 +76,12 @@ class HomePageFragment : WolmoFragment<HomePagePresenter>(), IHomePageView {
         private const val NEWS_INACTIVE_TAB = R.drawable.ic_news_list_off
         private const val PROFILE_ACTIVE_TAB = R.drawable.ic_profile_on
         private const val PROFILE_INACTIVE_TAB = R.drawable.ic_profile_off
+        private const val TITLE_NEWS = "News"
+        private const val TITLE_PROFILE = "Profile"
+    }
+
+    override fun onSelectedViewPager(position: Int) {
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/HomePageFragment.kt]: Companion object must be in the bottom of the class. Move this method above it

File: app/src/main/res/layout/fragment_profile.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent">
+
+    <ImageView
+        android:id="@+id/imageView3"
```
Comment (lmiotti) [inline app/src/main/res/layout/fragment_profile.xml]: Can we use a more descriptive id for this?

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        fun bindTo(new: New) {
+            itemView.let {
+                it.vNewTitle.text = new.title
+                it.vNewContent.text = new.content
+            }
+        }
+    }
+}
+
+class DiffCallback : DiffUtil.ItemCallback<New>() {
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt]: This is perfect, but can we move it to a new file?

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): NewsViewHolder {
+        return NewsViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item_new, parent, false))
+    }
+
+    override fun onBindViewHolder(holder: NewsViewHolder, position: Int) {
+        holder.bindTo(newsItem[position])
+    }
+
+    fun addAllNews(news: List<New>) {
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt]: This is not really necesary, I'm wrong? In fragment you can do submitList and it's done with that.

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        newsItem = news
+        notifyDataSetChanged()
+    }
+
+    override fun getItemCount(): Int {
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt]: Same as above. You can remove this, if it's added for a particular purpose please tell me

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import ar.com.wolox.android.R
+import ar.com.wolox.android.example.model.New
+import kotlinx.android.synthetic.main.item_new.view.*
+
+class NewsAdapter : ListAdapter<New, NewsAdapter.NewsViewHolder>(DiffCallback()) {
+
+    private var newsItem = listOf<New>()
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt]: If addAllNews is removed we can erase this too

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt [hunk 7]
```diff
@@ -2,16 +2,24 @@ package ar.com.wolox.android.example.ui.home.news
...
+import androidx.recyclerview.widget.RecyclerView
 import ar.com.wolox.android.R
+import ar.com.wolox.android.example.model.New
 import ar.com.wolox.wolmo.core.fragment.WolmoFragment
 import kotlinx.android.synthetic.main.fragment_news.*
 import javax.inject.Inject
 
 class NewsFragment @Inject constructor() : WolmoFragment<NewsPresenter>(), INewsView {
 
+    private lateinit var mNewsAdapter: NewsAdapter
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt]: In kotlin we are not using anymore 'm' as prefix

**Review by lmiotti — CHANGES_REQUESTED** [review]
More things to be changed. In general is very good.

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt (hunk 5 above)
Comment (rickwolox) [reply app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsAdapter.kt]: Without this line the onBindViewHolder is not called

File: app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    override fun setLoadingProgressBarVisible() {
+        vNewsSwipeContainer.isRefreshing = true
+    }
+
+    override fun setLoadingProgressBarGone() {
+        vNewsSwipeContainer.isRefreshing = false
+    }
+
+    override fun showNews(news: List<New>) {
+        mNewsAdapter.addAllNews(news)
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/home/news/NewsFragment.kt]: Use `newsAdapter.submitList(news)`

**Review by lmiotti — APPROVED** [review]
awesome2

**Pull request merged**